src/cargo/core/compiler/fingerprint.rs

```diff
@@ -1600,28 +1625,26 @@ pub fn parse_dep_info(
     pkg_root: &Path,
     target_root: &Path,
     dep_info: &Path,
-) -> CargoResult<Option<Vec<PathBuf>>> {
+) -> CargoResult<Option<RustcDepInfo>> {
     let data = match paths::read_bytes(dep_info) {
         Ok(data) => data,
         Err(_) => return Ok(None),
     };
-    let paths = data
-        .split(|&x| x == 0)
-        .filter(|x| !x.is_empty())
-        .map(|p| {
-            let ty = match DepInfoPathType::from_byte(p[0]) {
-                Some(ty) => ty,
-                None => return Err(internal("dep-info invalid")),
-            };
-            let path = util::bytes2path(&p[1..])?;
-            match ty {
-                DepInfoPathType::PackageRootRelative => Ok(pkg_root.join(path)),
-                // N.B. path might be absolute here in which case the join will have no effect
-                DepInfoPathType::TargetRootRelative => Ok(target_root.join(path)),
-            }
-        })
-        .collect::<Result<Vec<_>, _>>()?;
-    Ok(Some(paths))
+    let info = match OnDiskDepInfo::parse(&data) {
+        Some(info) => info,
+        None => return Ok(None),
```

Comment:
Could this at least log a warning that it failed to parse the dep-info file?  I assume this would only happen if the file is corrupt (or in some different format?).  I assume you changed this so if the format is changed again in the future, cargo will just silently rebuild (which seems like the correct approach).

Reply:
Hm actually I misinterpreted the previous code. I thought the previous iteration also ignored parsing errors but it looks like it only ignores I/O and/or not present errors. I do think, though, that it's reasonable to ignore corrupt files here since we'll rewrite them later anyway.